radicle-router.md

```diff
@@ -0,0 +1,7 @@
+# Radicle Router
+
+## Nice to haves:
+- Should allow nesting of routes inside App.svelte or Routes files
+- Should be able to parse routes without hard typing them all
+- Preloading of components
+-
```

Comment:
This got committed by accident :sweat_smile: 
But yeah I'm thinking to use svelte actions kinda like `<a href="" use:link>Somewhere</a>` This way we don't have to add `on:click` and have still programatic navigation.

Reply:
Why would we need the `use:link` action though? wouldn't a regular function that converts the route be enough? I.e. `<a href={routeToPath({...})} >`.

Reply:
> Why would we need the `use:link` action though? wouldn't a regular function that converts the route be enough? I.e. `<a href={routeToPath({...})} >`.

The issue with this approach is that the browser would reload the entire application on each navigation, to make the app feel like a SPA without the browser reloading the entire page on each anchor tag, we need to replace the `onClick` behavior of the anchor tag and the nicest way IMO is to use this `use:link` approach.

Reply:
Hmm. Why would the browser reload the whole page though, if we only replace the `#` hash part?

Reply:
> Hmm. Why would the browser reload the whole page though, if we only replace the `#` hash part?

Thats because we are not replacing the hash part but the whole pathname.
One nice thing would be to have different navigation modes, e.g. https://github.com/AlexxNB/tinro#navigation-method
But I think I'll leave that for another PR.

Reply:
But using the history API and making the routing hash-based aren't mutually exclusive. I think we should consider hash-based routing. It would make it easier to use plain anchor tags and the middle-click functionality would work out of the box.

Reply:
Not a hill I'll die on, didn't want to also change the navigation method while introducing a in house router..
wdyt re: using hash routing @cloudhead?

It would also allow us to deploy on IPFS

Reply:
Yeah, on a second thought, maybe we punt on this for now.